codex/sales/states/downloading.nim

```diff
@@ -38,5 +41,24 @@ method run*(state: SaleDownloading, machine: Machine): Future[?State] {.async.}
   without request =? data.request:
     raiseAssert "no sale request"
 
+  without availability =? await context.reservations.find(
+      request.ask.slotSize,
+      request.ask.duration,
+      request.ask.pricePerSlot,
+      used = false):
+    info "no availability found for request, ignoring",
+      slotSize = request.ask.slotSize,
+      duration = request.ask.duration,
+      pricePerSlot = request.ask.pricePerSlot,
+      used = false
+    return
+  
+  data.availability = some availability
+
+  if err =? (await agent.context.reservations.markUsed(
+    availability,
+    request.slotId(data.slotIndex))).errorOption:
+    raiseAssert "failed to mark availability as used, error: " & err.msg
```

Comment:
This would crash the node, we should probably move to the error state instead.

Reply:
done